pkg/graph/schema.graphql

```diff
@@ -1144,6 +1187,7 @@ type Query {
   currentUser: CurrentUser
   cedarSystem(id: String!): CedarSystem
   cedarSystems: [CedarSystem]
+  deployments(systemId: String!, deploymentType: String, state: String, status: String): [CedarDeployment!]!
```

Comment:
Should we (likely in addition to having a `deployments` query) make a `deployments` array be a property of a `cedarSystem`?

I know this would give us some crazy n+1 queries if we got ALL cedar systems and ALL deployments for those systems - but would it make sense from a schema perspective to have this?

If we did that, would the entire query take the time for EASi to get the System (1 round trip total), then the time for EASi to get the deployment (now 2 round trips), rather than just getting the deployment directly?

Doing a lot of thinking out loud here, but curious on peopls thoughts!

Reply:
On one hand, it's a nice addition to the GraphQL schema. On the other hand, yeah, there's some serious n+1 issues if you query for multiple systems.

Looking at and testing the CEDAR API, there's nothing _documented_ about passing multiple system IDs to the `/deployments` endpoint...but you can pass multiple `systemId` query parameters to the query. It looks like the API's support for that is buggy (or accidental/unintended); the response contains a `systemIdList` with all the system IDs passed in, but the only deployment data returned is for the first systemID specified.

Reply:
I'm inclined to just have `CedarDeployment` be separate for right now; we can make a ticket for later work to add it to the `CedarSystem` type, depending on the response from the CEDAR team.

Reply:
Agree. Lets keep it split and just have the UI make the queries manually. Not the ideal GQL setup, but without a solve for n+1, I think it makes the most sense.